### DISCUSSION TITLE
failing to compile contact module

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#19569] [Q&A Modules: General] [answered]
Dear all,
after a fresh update of libmesh and moose, I cannot compile the contact module any longer. Errors seem to be related to some malfunctions in the TIMPI libmesh - no type (buffer_type) and accessing private members or not defined functions. 
I should add that other modules (and the framework) compile just fine, and that my moose-libmesh version (via conda) is somewhat older (no idea why, since I also update conda)  than the one posted in the discussions (2021.10.27 vs 2021.11.11 --> might this do the trick), and I am running on an ubuntu 18.04 LTS system. 
Any help/hints would be great, thanks,
mauro

**Comment by GiudGiud** [comment]
Hello

Your moose-libmesh version cannot be older than the supported one, otherwise moose cant compile.
Please follow the instructions on that post you saw

Guillaume

**Reply by mcacace** [reply]
@GiudGiud I did follow the steps for updating via conda, but cannot upgrade my moose-libmesh version further.

**Comment by ngrilli** [comment, marked as answer]
Dear @mcacace 

I updated last week and also has problems compiling.
I ended up cancelling completely the moose folder and starting the Ubuntu installation procedure from scratch,
so that also libmesh and petsc get updated.

Best Regards,
Nicolò

**Reply by mcacace** [reply]
@ngrilli thanks! I would try to forced removing the moose envs in the conda and reinstall all. BTW, did you also upgrade to mamba? And, is it worth? Thanks

**Reply by GiudGiud** [reply]
see this post
https://github.com/idaholab/moose/discussions/19532
This is the current solution to update

**Reply by mcacace** [reply]
@ngrilli It worked, I have now the current libmesh version (and wow, mamba is indeed faster!).

**Reply by mcacace** [reply]
Thanks @GiudGiud -- I should have read the whole post ...
